src/org/opensolaris/opengrok/configuration/messages/ProjectMessage.java

```diff
@@ -248,6 +273,60 @@ private void validateMore(RuntimeEnvironment env) throws Exception {
 
                 env.refreshDateForLastIndexRun();
                 break;
+            case "set":
+                Matcher matcher = VARIABLE_PATTERN.matcher(getText().substring(4));
+                if (!matcher.find()) {
+                    // invalid pattern
+                    throw new IOException(
+                            String.format("The pattern \"%s\" does not match \"%s\".",
+                                    VARIABLE_PATTERN.toString(),
+                                    getText()));
+                }
+                        
+                for (String projectName : getTags()) {
+                    Project project;
+                    if ((project = env.getProjects().get(projectName)) != null) {
+                        // Set the property.
+                        ClassUtil.invokeSetter(
+                                project,
+                                matcher.group(1), // field
+                                matcher.group(2) // value
+                        );
+                        
+                        // Refresh repositories for this project as well.
+                        List<RepositoryInfo> riList = env.getProjectRepositoriesMap().get(project);
+                        if (riList != null) {
+                            for (RepositoryInfo ri : riList) {
+                                Repository repo = getRepository(ri, false);
+                                
+                                // set the property
+                                ClassUtil.invokeSetter(
+                                        repo,
+                                        matcher.group(1), // field
+                                        matcher.group(2) // value
+                                );
+                            }
+                        }
+                        
+                        return String.format("Variable \"%s\" set to \"%s\".",
+                                matcher.group(1), matcher.group(2)).getBytes();
+                    } else {
+                        LOGGER.log(Level.WARNING, "cannot find project " +
+                               projectName + " to set a property");
+                    }
+                }
+                break;
+            case "get":
+                for (String projectName : getTags()) {
+                    Project project;
+                    if ((project = env.getProjects().get(projectName)) != null) {
+                        return ClassUtil.invokeGetter(project, getText().substring(4)).getBytes();
```

Comment:
I'm worried that this will set just the first project in getTags().

Reply:
There is a check that `get` can only have one project specified.

Reply:
I didn't see that, ok.